### PR TITLE
Issue #20985 Don't mount the docker/config.json if file doesn't exist

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

**Comment by praveenkumar** [comment]
cc @deads2k

File: pkg/oc/clusterup/run_self_hosted.go [hunk 1]
```diff
@@ -529,8 +529,12 @@ func (c *ClusterUpConfig) startKubelet(out io.Writer, masterConfigDir, nodeConfi
 	} else if currentUser, err := user.Current(); err == nil {
 		var cfgPath string
 		cfgPath = filepath.Join(currentUser.HomeDir, ".docker", "config.json")
-		container.ContainerBinds = append(container.ContainerBinds, fmt.Sprintf("%s:/root/.docker/config.json", cfgPath))
-		glog.Infof("docker config path %s", cfgPath)
+		// First check if config file exist on the host before add it to the mounts.
+		// https://github.com/openshift/origin/issues/20985
+		if _, err := os.Stat(cfgPath); err == nil {
+			container.ContainerBinds = append(container.ContainerBinds, fmt.Sprintf("%s:/root/.docker/config.json", cfgPath))
+			glog.Infof("docker config path %s", cfgPath)
```
Comment (mfojtik) [inline pkg/oc/clusterup/run_self_hosted.go]: drop

File: pkg/oc/clusterup/run_self_hosted.go [hunk 2]
```diff
@@ -529,8 +529,12 @@ func (c *ClusterUpConfig) startKubelet(out io.Writer, masterConfigDir, nodeConfi
 	} else if currentUser, err := user.Current(); err == nil {
 		var cfgPath string
 		cfgPath = filepath.Join(currentUser.HomeDir, ".docker", "config.json")
-		container.ContainerBinds = append(container.ContainerBinds, fmt.Sprintf("%s:/root/.docker/config.json", cfgPath))
-		glog.Infof("docker config path %s", cfgPath)
+		// First check if config file exist on the host before add it to the mounts.
+		// https://github.com/openshift/origin/issues/20985
```
Comment (mfojtik) [inline pkg/oc/clusterup/run_self_hosted.go]: do not reference issues links in comments

File: pkg/oc/clusterup/run_self_hosted.go [hunk 3]
```diff
@@ -529,8 +529,12 @@ func (c *ClusterUpConfig) startKubelet(out io.Writer, masterConfigDir, nodeConfi
...
+		if _, err := os.Stat(cfgPath); err == nil {
```
Comment (mfojtik) [inline pkg/oc/clusterup/run_self_hosted.go]: os.IsExists(err)

Comment (praveenkumar) [reply pkg/oc/clusterup/run_self_hosted.go]: Done

**Comment by praveenkumar** [comment]
@mfojtik made changes as per suggestion.

**Comment by praveenkumar** [comment]
@mfojtik this also need cherry pick for 3.11 branch.

**Comment by mfojtik** [comment]
/cherrypick 3.11

@derekwaynecarr can we have this in 3.11 ?

File: pkg/oc/clusterup/run_self_hosted.go [hunk 4]
```diff
@@ -529,8 +529,10 @@ func (c *ClusterUpConfig) startKubelet(out io.Writer, masterConfigDir, nodeConfi
 	} else if currentUser, err := user.Current(); err == nil {
 		var cfgPath string
 		cfgPath = filepath.Join(currentUser.HomeDir, ".docker", "config.json")
-		container.ContainerBinds = append(container.ContainerBinds, fmt.Sprintf("%s:/root/.docker/config.json", cfgPath))
-		glog.Infof("docker config path %s", cfgPath)
+		// First check if config file exist on the host before add it to the mounts.
+		if _, err := os.Stat(cfgPath); os.IsExist(err) {
```
Comment (derekwaynecarr) [inline pkg/oc/clusterup/run_self_hosted.go]: is this right?

shouldnt it check `if _, err := os.Stat(path); err == nil {`

Comment (smarterclayton) [reply pkg/oc/clusterup/run_self_hosted.go]: Yes

**Pull request merged**